### ISSUE TITLE
Consider combining EntityEqualityRewritingExpressionVisitor with NavigationRewritingExpressionVisitor

### ISSUE THREAD

**Issue by AndriySvyryd** [dotnet/efcore#8243]
Without `EntityEqualityRewritingExpressionVisitor` running beforehand `NavigationRewritingExpressionVisitor` doesn't rewrite optional navigation equalities correctly.

**Comment by smitpatel** [comment]
Example?
If we keep things separate that NavRewrite should only convert nav access to joined queries hence producing new qsre. Entityequality rewrite should be step afterwards so that it can also rewrite new qsre representing entities.

**Comment by AndriySvyryd** [comment]
@smitpatel  Try commenting out the line
```
queryModel.TransformExpressions(_entityEqualityRewritingExpressionVisitor.Visit);
```
in `QueryOptimizer` and run `ComplexNavigationsQueryInMemoryTest`

**Comment by smitpatel** [comment]
The awkward dependency. Navigation rewrite should be independent of entity equality rewrite.

**Comment by smitpatel** [comment]
This should be fixed when working on #7773

**Comment by smitpatel** [comment]
Closing as won't fix since we want to keep them separate in new pipeline for better modularity. We can revisit the decision if run into issues.
